lib/auth/auth.go

```diff
@@ -1137,6 +1143,117 @@ func (a *Server) GetMFAAuthenticateChallenge(user string, password []byte) (*MFA
 	return chal, nil
 }
 
+// CreateAuthenticationChallenge creates and returns challenges for a users MFA devices.
+func (a *Server) CreateAuthenticationChallenge(ctx context.Context, req *proto.CreateAuthenticationChallengeRequest) (*proto.MFAAuthenticateChallenge, error) {
+	username := ""
+
+	// Determine how to verify user.
+	switch req.GetRequest().(type) {
+	case *proto.CreateAuthenticationChallengeRequest_UserPwd:
+		username = req.GetUserPwd().Username
+		err := a.WithUserLock(username, func() error {
+			return a.checkPasswordWOToken(username, req.GetUserPwd().Password)
+		})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+	case *proto.CreateAuthenticationChallengeRequest_TokenID:
+		token, err := a.getRecoveryStartToken(ctx, req.GetTokenID())
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		username = token.GetUser()
+
+	default:
+		return nil, trace.BadParameter("missing request parameter")
+	}
+
+	protoChal, err := a.mfaAuthChallenge(ctx, username, a.Identity)
```

Comment:
nit: Add a comment on the a.Identity parameter, otherwise it's not clear why it needs to be there (as mfaAuthChallenge has access to `a`).

```suggestion
	protoChal, err := a.mfaAuthChallenge(ctx, username, a.Identity /* u2fStorage */)
```